threading/channels.nim

```diff
@@ -100,7 +100,7 @@ runnableExamples("--threads:on --gc:orc"):
 when not (defined(gcArc) or defined(gcOrc) or defined(gcAtomicArc) or defined(nimdoc)):
   {.error: "This module requires one of --mm:arc / --mm:atomicArc / --mm:orc compilation flags".}
 
-import std/[locks, isolation, atomics]
+import std/[locks, isolation, atomics, options]
```

Comment:
Nah. I don't like `options` and nothing about this new API is hardly any better than the old one:

```nim

var x: T
if tryRecv(chan, x): ...
```

vs:

```nim

let x = tryRecv(chan) 
if x.isSome: ...

```

Reply:
fair enough